src/DynamicExpresso.Core/Parsing/Parser.cs

```diff
@@ -1961,6 +1969,16 @@ private static bool CheckIfMethodIsApplicableAndPrepareIt(MethodData method, Exp
 				var paramsArrayElementType = paramsArrayTypeFound.GetElementType();
 				if (paramsArrayElementType == null)
 					throw new Exception("Type is not an array, element not found");
+
+				if (paramsArrayElementType.IsGenericParameter)
+				{
+					var actualTypes = paramsArrayPromotedArgument.Select(_ => _.Type).Distinct().ToArray();
+					if (actualTypes.Length != 1)
+						throw new Exception($"The type arguments for method '{method.MethodBase}' cannot be inferred from the usage.");
```

Comment:
I would have preferred to throw a `ParserException` instead of the current `Exception`, but it's really far down the stack, and that would require passing the token position all the way down, which didn't seem great.

Reply:
@metoule 
Maybe it is better to throw a `ParseException` anyway with a -1 position? Just because in this way a caller can catch only it and avoid using a generic `Exception`. Here and also some line above ...

What do you think?

Reply:
That's a good idea, I'll update the PR! Thanks!